lib/matplotlib/artist.py

```diff
@@ -147,6 +147,23 @@ def remove(self):
         # callback has one parameter, which is the child to be removed.
         if self._remove_method is not None:
             self._remove_method(self)
```

Comment:
Surely the logic for removing from the axes should live in the remove method itself. Is there a reason we can't do that?